src/openforms/submissions/views.py

```diff
@@ -235,3 +240,41 @@ def get_sendfile_opts(self) -> dict:
             "mimetype": submission_attachment.content_type,
         }
         return opts
+
+
+class SearchSubmissionForCosignFormView(PermissionRequiredMixin, FormView):
+    form_class = SearchSubmissionForCosignForm
+    template_name = "submissions/find_submission_for_cosign.html"
+    raise_exception = True
+
+    cosign_url = None
+
+    def has_permission(self):
+        """
+        The user should have authenticated with the auth plugin specified on the form for the co-sign component
+        """
+        form = get_object_or_404(Form, slug=self.kwargs["form_slug"])
+        cosign_component = form.get_cosign_component()
+        expected_auth_plugin = cosign_component["authPlugin"]
+        return is_authenticated_with_plugin(self.request, expected_auth_plugin)
```

Comment:
I think using `UserPassesTestMixin` instead of `PermissionRequiredMixin` is more appropriate here - the latter really typically refers to codename permissions like `"foo.can_bar"`. Given that you need to override the main implementation (`has_permission`) to make it work, I think the `def test_func` of the former mixin is more appropriate.